app/src/main/java/org/simple/clinic/bp/BloodPressureRepository.kt

```diff
@@ -54,6 +56,19 @@ class BloodPressureRepository @Inject constructor(
         }
   }
 
+  fun updateMeasurement(bloodPressureMeasurement: BloodPressureMeasurement): Completable {
+    return Single
+        .fromCallable {
+          bloodPressureMeasurement.copy(
+              updatedAt = Instant.now(clock),
+              syncStatus = SyncStatus.PENDING
+          )
+        }
+        .flatMapCompletable { updatedBloodPressureMeasurement ->
+          Completable.fromAction { dao.save(listOf(updatedBloodPressureMeasurement)) }
+        }
+  }
```

Comment:
I think `Single.fromCallable` isn't required here and this can be made simpler:

```kotlin
return Completable.fromAction {
  val updatedMeasurement = bloodPressureMeasurement.copy(
      updatedAt = Instant.now(clock),
      syncStatus = SyncStatus.PENDING)
  dao.save(listOf(updatedMeasurement))
}
```

What do you think?

Reply:
Oh yeah, it isn't necessary. It's just a habit where I want to make sure each operator in an Rx chain does only one thing. I don't mind changing it to this format because the chain is small.

Reply:
I understand and I think that works very nicely for single line lambdas.